### PR TITLE
(feat): ignore certain fields for configuration drift evaluation

### PR DESCRIPTION
**Pull request by gianlucam76**
When creating a Profile/ClusterProfile with syncMode set to `ContinuosWithDriftDetection`, user has the ability to specify a set of fields, for certain resources, to be ignored when evaluation configuration drift.

For example, consider a Deployment managed by an autoscaler. The autoscaler dynamically adjusts replica count based on workload. In this scenario, it's crucial to distinguish between intentional deployment changes (e.g., deletion) and expected replica fluctuations. By ignoring replica count changes, Sveltos can accurately detect meaningful configuration drifts.

Drift-detection-manager will finally consume those Patches before evaluating configuration drift.

This PR adds the ability in drift-detection-manager to get the information on which fields need to be ignored during configuration drift evaluation.

### CODE REVIEW

File: controllers/resourcesummary_controller_test.go [hunk 1]
```diff
@@ -184,6 +186,10 @@ var _ = Describe("ResourceSummary Reconciler", func() {
 		Expect(controllers.UpdateMapsAndResourceSummaryStatus(reconciler, context.TODO(), resourceSummary, logger)).To(Succeed())
 
 		resourceSummaryRef := getObjRefFromResourceSummary(resourceSummary)
+
```
Comment (kprav33n) [inline controllers/resourcesummary_controller_test.go:189]: Should this block added for debugging should be removed?

**Review by kprav33n — APPROVED** [review]
LGTM. Thank you.

File: controllers/resourcesummary_controller_test.go (hunk 1 above)
Comment (gianlucam76) [reply controllers/resourcesummary_controller_test.go:189]: Yes thank you for catching it. Will amend and remove

Comment (gianlucam76) [reply controllers/resourcesummary_controller_test.go:189]: Thanks @kprav33n. Abhinav mentioned he is also almost done. So i m happy to drop this and use his PR instead.

Comment (gianlucam76) [reply controllers/resourcesummary_controller_test.go:187]: Discussed with Abhinav and merging this with his blessing.

**Pull request merged**